internal/frameworkprovider/service_resource_model.go

```diff
@@ -1,64 +1,104 @@
-	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
 	v1alphaService "github.com/nobl9/nobl9-go/manifest/v1alpha/service"
 )
 
 // ServiceResourceModel describes the [ServiceResource] data model.
 type ServiceResourceModel struct {
-	Name        string            `tfsdk:"name"`
-	DisplayName types.String      `tfsdk:"display_name"`
-	Project     string            `tfsdk:"project"`
-	Description types.String      `tfsdk:"description"`
-	Annotations map[string]string `tfsdk:"annotations"`
-	Labels      Labels            `tfsdk:"label"`
-	Status      types.Object      `tfsdk:"status"`
+	Name             string                 `tfsdk:"name"`
+	DisplayName      types.String           `tfsdk:"display_name"`
+	Project          string                 `tfsdk:"project"`
+	Description      types.String           `tfsdk:"description"`
+	Annotations      map[string]string      `tfsdk:"annotations"`
+	Labels           Labels                 `tfsdk:"label"`
+	ResponsibleUsers []ResponsibleUserModel `tfsdk:"responsible_users"`
+	ReviewCycle      *ReviewCycleModel      `tfsdk:"review_cycle"`
+}
+
+// ResponsibleUserModel represents [v1alphaService.ResponsibleUser].
+type ResponsibleUserModel struct {
+	ID types.String `tfsdk:"id"`
 }
 
-type ServiceResourceStatusModel struct {
-	SLOCount types.Int64 `tfsdk:"slo_count"`
+func (r ResponsibleUserModel) ToManifest() v1alphaService.ResponsibleUser {
+	return v1alphaService.ResponsibleUser{ID: r.ID.ValueString()}
 }
 
-func newServiceResourceConfigFromManifest(
-	ctx context.Context,
-	svc v1alphaService.Service,
-) (*ServiceResourceModel, diag.Diagnostics) {
-	var status types.Object
-	statusType, diags := serviceResourceSchema.TypeAtPath(ctx, path.Root("status"))
-	if diags.HasError() {
-		return nil, diags
+// sortResponsibleUsers sorts the API returned list based on the user-defined list as a reference for sorting order.
+func sortResponsibleUsers(userDefinedResponsibleUsers, apiReturnedList []ResponsibleUserModel) []ResponsibleUserModel {
+	// Preserve nil when the user-defined list is nil and API returns an empty list.
+	// This ensures consistency between null and empty list in Terraform state.
+	if userDefinedResponsibleUsers == nil && len(apiReturnedList) == 0 {
+		return nil
 	}
-	statusAttrs := statusType.(basetypes.ObjectType).AttrTypes
-	if svc.Status != nil {
-		statusModel := ServiceResourceStatusModel{
-			SLOCount: types.Int64Value(int64(svc.Status.SloCount)),
-		}
-		v, diags := types.ObjectValueFrom(ctx, statusAttrs, statusModel)
-		if diags.HasError() {
-			return nil, diags
-		}
-		status = v
-	} else {
-		status = types.ObjectNull(statusAttrs)
+	return sortListBasedOnReferenceList(
+		apiReturnedList,
+		userDefinedResponsibleUsers,
+		func(a, b ResponsibleUserModel) bool {
+			return a.ID == b.ID
+		},
+	)
+}
+
+type ReviewCycleModel struct {
+	RRule     types.String `tfsdk:"rrule"`
+	StartTime types.String `tfsdk:"start_time"`
+	TimeZone  types.String `tfsdk:"time_zone"`
```

Comment:
for non-nullable (required) primitive types you don't need to use `types` package:


```suggestion
	RRule     string `tfsdk:"rrule"`
	StartTime string `tfsdk:"start_time"`
	TimeZone  string `tfsdk:"time_zone"`
```